graphql_compiler/compiler/ir_lowering_match/utils.py

```diff
@@ -222,12 +222,17 @@ def construct_where_filter_predicate(simple_optional_root_info):
 CompoundMatchQuery = namedtuple('CompoundMatchQuery', ('match_queries'))
 
 
-class OptionalTraversalTrie(dict):
-    def __init__(self):
+class OptionalTraversalTrie():
+    def __init__(self, complex_optional_roots):
         """Initialize empty trie of optional root Locations (elements of complex_optional_roots)."""
-        super(OptionalTraversalTrie, self).__init__()
+        self._location_to_children = {
+            optional_root_location: set()
+            for optional_root_location in complex_optional_roots
+        }
+        self._root_location = Location(())
```

Comment:
Since this location is a dummy, I'd make it `None` instead of such an invalid `Location` object. Then you can define it as a class-level constant.